### ISSUE TITLE
Explainability output enhancements 

### ISSUE THREAD

**Issue by adisos** [np-guard/vpc-network-config-analyzer#387]
Example for current output: 
```
Connectivity explanation for protocol: TCP between vsi1-ky[10.240.10.4] and 161.26.0.0/16
===================================================================================================

Connection protocol: TCP exists between vsi1-ky[10.240.10.4] and Public Internet 161.26.0.0/16
External Router PublicGateway: public-gw-ky
Egress: security group sg1-ky allows connection (with allow rules); network ACL acl1-ky allows connection (with allow rules)
```

I thought we could improve the output by presenting the actual "path" of the relevant resources/configurations affecting the connectivity, by the relevant order. 

so for the case above, it may be like:
```
vsi1-ky[10.240.10.4] ->  SG sg1-ky ->  ACL acl1-ky -> PublicGateway: public-gw-ky ->  Public Internet 161.26.0.0/16
```

or also with the subnet:
```
vsi1-ky[10.240.10.4] ->  SG sg1-ky -> subnet ... ->  ACL acl1-ky -> PublicGateway: public-gw-ky ->  Public Internet 161.26.0.0/16
```

For denied connectivity, could be a prefix of the path until the part that blocks the connectivity. 

@ShiriMoran @kyorav @zivnevo  what do you think?

**Comment by kyorav** [comment]
I think it's a very good idea. It makes it much easier to understand the explanation and removes a lot of unnecessary verbiage. 
In terms of format, there are a few corner cases we need to figure out. For example, if there are multiple SGs allowing the connection on the same endpoint we need to list them all in the same "step". So maybe put them in brackets?
```
vsi1-ky[10.240.10.4] ->  SG [sg1-ky, sg2-ky] -> subnet ...
```
Also, this puts everything in one long line, which can be hard to follow and will look differently in different displays (with/without line wrap of different size). Maybe we can put each step in a separate line, and then we can also use indentation to convey some information
```
vsi1-ky[10.240.10.4] ->  
    SG sg1-ky ->  
    ACL acl1-ky -> 
    PublicGateway: public-gw-ky ->  
Public Internet 161.26.0.0/16
```

**Comment by zivnevo** [comment]
I think this will become really useful as we add more and more network devices (RTs, LBs) and as we move to multi-cloud scenarios.
However, I don't want to delay v0.3 for this feature. Maybe make it for v0.4 or even v0.5.

**Comment by ShiriMoran** [comment]
> For denied connectivity, could be a prefix of the path until the part that blocks the connectivity.
> 
> @ShiriMoran @kyorav @zivnevo what do you think?

@adiso I agree this presentation will be helpful for allowed connectivity. 
Not so sure about denied connectivity. There are two issues here
1. Providing information only until the first part that blocks the connectivity does not seem right. We should provide all available information. Plus I don't find it very consistent - consider the path
`vsi1 -> sg1 -> nacl1 -> nacl2 -> sg2 -> vsi2`
If e.g. `sg1` and `sg2` are blocking we will only inform the user that `sg1` is blocking. It doesn't make any sense to me.
1. Drawing the path in cases it is blocked seems less intuitive here, especially given the above. Sticking to the path also in the deny case will require a consistent way of providing information after the first block.

**Comment by ShiriMoran** [comment]
Solved by PR #435
Thoughts, comments etc welcome @adisos  @zivnevo @kyorav 

1. Examples of path when connection exists:
vsi to vsi different subnets:
```
The following connection exists between vsi1-ky[10.240.10.4] and vsi2-ky[10.240.20.4]: protocol: TCP,UDP
Path:
	vsi1-ky[10.240.10.4] -> security group sg1-ky -> subnet1-ky -> network ACL acl1-ky -> 
	network ACL acl2-ky -> subnet2-ky -> security group sg1-ky -> vsi2-ky[10.240.20.4]
```

```
vsi to vsi same subnets:
The following connection exists between vsi3b-ky[10.240.30.6] and vsi3a-ky[10.240.30.5]: All Connections
Path:
	vsi3b-ky[10.240.30.6] -> security group sg1-ky -> 
	security group sg1-ky -> vsi3a-ky[10.240.30.5]
```

vsi to external:
```
The following connection exists between vsi1-ky[10.240.10.4] and Public Internet 161.26.0.0/16: protocol: UDP
Path:
	vsi1-ky[10.240.10.4] -> security group sg1-ky -> subnet1-ky -> network ACL acl1-ky -> 
	PublicGateway public-gw-ky -> 
	Public Internet 161.26.0.0/16
```
2. Examples of path when egress/ingress blocks path. In this case I also print the former detailed header to provide full information regarding the tables and router

```
No connection between vsi1-ky[10.240.10.4] and vsi3a-ky[10.240.30.5]; connection blocked by egress
Egress: security group sg1-ky allows connection; network ACL acl1-ky blocks connection
Ingress: network ACL acl3-ky allows connection; security group sg1-ky allows connection
Path:
	vsi1-ky[10.240.10.4] -> security group sg1-ky -> subnet1-ky -> |
```

```
No connection between vsi1-ky[10.240.10.4] and Public Internet 100.128.0.0/32; connection blocked by egress
External traffic via PublicGateway: public-gw-ky
Egress: security group sg1-ky allows connection; network ACL acl1-ky blocks connection
Path:
	vsi1-ky[10.240.10.4] -> security group sg1-ky -> subnet1-ky -> 
```

3. In case the connection is to/from public internet and a router was not defined I am not printing the path which is also non-defined (is nacl part of the path)? and the printing did not change

**Comment by adisos** [comment]
Initial version was merged.
@kyorav @zivnevo  please experiment with the new output, can also view example output from tests at https://github.com/np-guard/vpc-network-config-analyzer/commit/2f1feeb213a390f2a698396208ddd49425b2a9cc

**Comment by ShiriMoran** [comment]
> I would like to have the external IP of a FIP printed in the path

solved by https://github.com/np-guard/vpc-network-config-analyzer/pull/472
